### PR TITLE
Separated out buy functions into buyGear, buyArmoire, and buyPotion

### PR DESCRIPTION
**Pull request by TheHollidayInn**
#6887 User ops buy refactor
### Changes

Move Buy to:
BuyGear
BuyPotion
BuyArmoire

---

UUID: 206039c6-24e4-4b9f-8a31-61cbb9aa3f66

### CODE REVIEW

**Comment by paglias** [comment]
@TheHollidayInn thanks for this! there's a change we can rename the original buy to buyGear, keep the 3 new routes but also a more general `buy` route that'll work like the old one (call the specific buyGear/buyPotion/buyArmoire from the general buy route?)

**Comment by TheHollidayInn** [comment]
Updated!

File: common/script/index.js [hunk 1]
```diff
@@ -259,7 +263,7 @@ api.wrap = function wrapUser (user, main = true) {
       releasePets: _.partial(importedOps.releasePets, user),
       releaseMounts: _.partial(importedOps.releaseMounts, user),
       releaseBoth: _.partial(importedOps.releaseBoth, user),
-      buy: _.partial(importedOps.buy, user),
```
Comment (paglias) [inline common/script/index.js:262]: buyPotion and buyArmoire are missing

**Comment by paglias** [comment]
Except for the comment, the only thing I feel like we should add is a dummy `common.ops.buy` op that simply call the new ops so we can keep using common.ops.buy on the client if we want to

**Comment by TheHollidayInn** [comment]
So, the buy route that I have added back will call the common.ops.buy and do the "switch" logic there?

**Comment by paglias** [comment]
I'd say so

**Comment by TheHollidayInn** [comment]
Updated!

**Pull request merged**